### PR TITLE
docs(pyroscope.ebpf): rewrite container id sections, include PID label

### PR DESCRIPTION
**Pull request by korniltsev**
#### PR Description
pyroscope.ebpf component is now capable of profiling not only containers but also processes

Update docs to reflect that.

#### PR Checklist

- [ ] CHANGELOG.md updated
- [x] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/flow/reference/components/pyroscope.ebpf.md [hunk 1]
```diff
@@ -95,16 +95,16 @@ can help you pin down a profiling target.
...
 | `__container_id__` | The container ID derived from target.                                                                                            |
 
-### Container ID
+### Targets 
 
-Each collected stack trace is then associated with a specified target from the targets list, determined by a
-container ID. This association process involves checking the `__container_id__`, `__meta_docker_container_id`,
-and `__meta_kubernetes_pod_container_id` labels of a target against the `/proc/{pid}/cgroup` of a process.
+One of the following special labels `__container_id__`, `__meta_docker_container_id`, and `__meta_kubernetes_pod_container_id`, 
+`__process_pid__` _must always_ be present in each target of `targets` and correspond to the container or process to profile.
```
Comment (ptodev) [inline docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: Should  `__meta_docker_container_id`, `__meta_kubernetes_pod_container_id`, and
`__process_pid__` be mentioned in the table above?

Comment (korniltsev) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: idk. The table above is the table of labels we inject in target inside pyroscope.ebpf component.  And we do not inject __meta_docker_container_id, __meta_kubernetes_pod_container_id, 
__process_pid__  , but expect them to be provided from discovery components

**Review by cyriltovena — APPROVED** [review]
LGTM

**Comment by korniltsev** [comment]
@clayton-cornell  @knylander-grafana ptal at docs when you have time

File: docs/sources/flow/reference/components/pyroscope.ebpf.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: @ptodev @korniltsev Tough call. Reading the text above the table, it's clearly stated that the labels listed are injected into the profiles. If I understand this properly, at least one of the special labels have to be in the final target. Are these labels defined elsewhere in relation to this component? Can we link to them?

Comment (knylander-grafana) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: If it's one of the special labels, should the list of labels be joined by "and" or "or"? And sounds like all of them need to be there.

Comment (korniltsev) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: >If it's one of the special labels, should the list of labels be joined by "and" or "or"? And sounds like all of them need to be there.

at least one of them

Comment (korniltsev) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: >If I understand this properly, at least one of the special labels have to be in the final target. 

correct

> Are these labels defined elsewhere in relation to this component?

`__process_pid__` is provided by `discovery.process`.  `__meta_docker_container_id` is provided by `discovery.docker`, `__meta_kubernetes_pod_container_id` is provided by `discovery.kubernetes.`

> Can we link to them?

Yes. Can you please suggest how?

Comment (clayton-cornell) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: @korniltsev I don't see these labels clearly defined anywhere in the docs except in places like the `discovery.kubernetes` and `discovery.kublet` topics.  Could we define it this way (matching how it's documented in elsewhere)?

```suggestion
One of the following special labels _must_ be included in each target of `targets` and the label must correspond to the container or process that is profiled:

* `__container_id__`: The container ID.
* `__meta_docker_container_id`: The ID of the Docker container.
* `__meta_kubernetes_pod_container_id`: The ID of the Kubernetes pod container.
* `__process_pid__` : The process ID.
```

We could possibly extend the descriptions a little to make it even more clear what they label.

Comment (knylander-grafana) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md:101]: I like your suggestion, Clayton.

Comment (korniltsev) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md]: I've applied the suggestion, but I dont see links to the relevant components documentation

Comment (clayton-cornell) [reply docs/sources/flow/reference/components/pyroscope.ebpf.md]: @korniltsev I dug through the docs and I don't see anything specific to link to. I duplicated the style and content used for a similar requirement in other similar topics. That should be enough unless you think we need more here @ptodev ?

**Review by clayton-cornell — APPROVED** [review]

**Comment by clayton-cornell** [comment]
@korniltsev It looks to me like everything is answered here so I'll get this merged.

**Comment by korniltsev** [comment]
\M/

**Pull request merged**